src/main/java/Duke.java

```diff
@@ -1,10 +1,181 @@
+
+import java.util.Scanner;
+
 public class Duke {
+
     public static void main(String[] args) {
+
+        // Welcome Message
+        PrintWelcomeMessage();
+
+        // Active Chat
+        ActiveChat();
+
+        // Goodbye Message
+        PrintGoodbyeMessage();
+    }
+
+    private static void ActiveChat() {
+        boolean isBye = false;
+        boolean isEmptyList = true;
+        String input;
+        Scanner in = new Scanner(System.in);
+        int counter = 0;
+        Task[] list = new Task[100];
+
+        while(!isBye){
+            //store input
+            input = in.nextLine();
+            //check if input is empty
+            CheckIsEmptyList(list);
+            //process input
+            if (input.equals("bye")){ //check if bye
+                isBye = true;
+            } else if (input.equals("list")) { //check if list
+                ProcessList(counter, list);
+            } else if (input.contains("done") ) { //check if done
+                ProcessDone(input, list);
+            } else {
+                counter = ProcessTasks(input, counter, list); //process tasks
+            }
+        }
+    }
+
+    private static void CheckIsEmptyList(Task[] list) {
+        boolean isEmptyList;
+        if(list[0] != null) {
+            isEmptyList = false;
+        }
+    }
+
+    private static void ProcessList(int counter, Task[] list) {
+        if (list[0] != null){
+            PrintListMessage(counter, list);
+        } else {
+            PrintListButEmptyMessage();
+        }
+    }
+
+    private static void ProcessDone(String input, Task[] list) {
+        int donePos = input.indexOf("done");
+        if (list[0] != null) {
+            if (input.length() < donePos + 5) {
+                PrintDoneButNotSpecificMessage(); //when input contains done but no number
+            } else {
+                ProcessDone(input, list, donePos); //when input contains done and specified number
+            }
+        } else {
+            PrintDoneButEmptyMessage(); //when input contains done but list is empty
+        }
+    }
+
+    private static void ProcessDone(String input, Task[] list, int donePos) {
+        String itemNumDone = input.substring(donePos + 5, donePos + 6);
+        int itemNum = Integer.parseInt(itemNumDone);
+        list[itemNum - 1].setDone();
+        PrintDoneMessage(list, itemNum);
+    }
+
+    private static int ProcessTasks(String input, int counter, Task[] list) {
+        if (input.contains("todo")) {
```

Comment:
It might be cleaner to replace the strings "todo", "deadline", etc. with  named constants of type `static final String`